Add endpoint to fetch filters in JSON format

jchristgit opened the pull request:
While this is an API endpoint consumed by the bot, keep it in the
`resources` app instead of the `api` app, as all the logic and data for
resources is contained within the `resources` app and we don't want to
start messing around with that.

The response format from the endpoint is as follows:

    {
        "topics": [
            "algorithms-and-data-structures",
            "data-science",
            "databases",
            "discord-bots",
            "game-development",
            "general",
            "microcontrollers",
            "security",
            "software-design",
            "testing",
            "tooling",
            "user-interface",
            "web-development",
            "other"
        ],
        "payment_tiers": [
            "free",
            "paid",
            "subscription"
        ],
        "type": [
            "book",
            "community",
            "course",
            "interactive",
            "podcast",
            "project-ideas",
            "tool",
            "tutorial",
            "video"
        ],
        "difficulty": [
            "beginner",
            "intermediate"
        ]
    }

Closes #710.

wookie184 reviewed (changes requested):
The response format here doesn't seem ideal. My understanding is that "filters" is the display name, and "valid_filters" is what goes in the URL(?), but the response doesn't link these together.

We could link them together in the structure, along the lines of this:
```
[
    {
      "name": "Difficulty",
      "query_param": "difficulty",
      "options": [
        {"name": "Beginner", "query_value": "beginner"},
        ...
      ],
      "icon": "fas fa-brain",
      "hidden": false,
    },
    ...
]
```

Alternatively we could maybe just drop the "valid_filters" section and document the transformation to get from the name to the url.

wookie184 reviewed (changes requested):
This seems to undo the fix for https://github.com/python-discord/site/issues/1209, I think it will cause performance issues.

jchristgit committed:
Add endpoint to fetch filters in JSON format

While this is an API endpoint consumed by the bot, keep it in the
`resources` app instead of the `api` app, as all the logic and data for
resources is contained within the `resources` app and we don't want to
start messing around with that.

The response format from the endpoint is as follows:

    {
        "topics": [
            "algorithms-and-data-structures",
            "data-science",
            "databases",
            "discord-bots",
            "game-development",
            "general",
            "microcontrollers",
            "security",
            "software-design",
            "testing",
            "tooling",
            "user-interface",
            "web-development",
            "other"
        ],
        "payment_tiers": [
            "free",
            "paid",
            "subscription"
        ],
        "type": [
            "book",
            "community",
            "course",
            "interactive",
            "podcast",
            "project-ideas",
            "tool",
            "tutorial",
            "video"
        ],
        "difficulty": [
            "beginner",
            "intermediate"
        ]
    }

Closes #710.

wookie184 reviewed (approved):
Not tested but LGTM
